website/docs/language/tests/mocking.mdx

```diff
@@ -0,0 +1,366 @@
+---
+page_title: Tests - Provider Mocking
+description: >-
+  Provide mocked data for providers and resources within `terraform test`.
+---
+
+# Mocks
+
+-> **Note:** Mocked providers and the associated syntax are available in Terraform v1.7.0 and later.
+
+Prior to Terraform v1.7.0, Terraform test authors were required to initialize and configure real Terraform providers for their tests. The `command = plan` setting could be used to avoid creating real infrastructure, but providers were still required to be correctly configured with the appropriate credentials and cloud accounts.
+
+In Terraform v1.7.0, we launched support for mocking providers and modules, along with resources and data sources, within the Terraform test framework. Test authors can now use mock (or fake) providers, and their associated resources within their tests, requiring no credentials or associated accounts to exist. Within the context of `terraform test`, a mocked provider or resource will generate fake data for all computed attributes that would normally be provided by the underlying provider APIs.
+
+Mocking functionality can only be used within the context of the [`terraform test` language](/terraform/language/tests). Readers of the mocking documentation should be aware of the surrounding testing syntax and language features.
+
+In addition, the more advanced features of the mocking framework require knowledge about the following Terraform provider features:
+
+- The difference between [attributes, nested attributes,](/terraform/plugin/framework/handling-data/attributes) and [blocks](https://developer.hashicorp.com/terraform/plugin/framework/handling-data/blocks).
+- Definitions of [optional, required, and computed attributes](/terraform/plugin/framework/handling-data/attributes/string#configurability).
+
+
+## Mock Providers
+
+Within Terraform test files, you can use a new block type: `mock_provider`.
+
+Mock providers return the same schema as the original provider and can be passed into your configuration under test in place of the matching provider. All resources and data sources retrieved by a mock provider will set the relevant values from the configuration, and generate fake data for any computed attributes.
+
+Mock providers can be used directly in place of traditional `provider` blocks, and they share the same global namespace. During the execution of the `terraform test` command, Terraform does not distinguish between a real and a mocked provider.
```

Comment:
You'd get a config error - the `mock_provider` block just can't be used in a regular Terraform file so it should never be available during regular `plan` / `apply` operations.

Reply:
That makes sense. In that case we probably don't need to say anything explicitly here, since the error will be perfectly clear on first plan, and any `mock_provider` blocks left in are likely to be by mistake.